lib/fluent/plugin/out_http.rb

```diff
@@ -21,6 +21,15 @@
 require 'fluent/plugin/output'
 require 'fluent/plugin_helper/socket'
 
+# patch Net::HTTP to support extra_chain_cert which was added in Ruby feature #9758.
+# see: https://github.com/ruby/ruby/commit/31af0dafba6d3769d2a39617c0dddedb97883712
+class Net::HTTP
+  SSL_IVNAMES << :@extra_chain_cert unless SSL_IVNAMES.include?(:@extra_chain_cert)
+  SSL_ATTRIBUTES << :extra_chain_cert unless SSL_ATTRIBUTES.include?(:extra_chain_cert)
+
+  attr_accessor :extra_chain_cert
+end
```

Comment:
With assuming SSL_ATTRIBUTES include :extra_chain_cert if SSL_IVNAMES include :@ extra_chain_cert, single condition can enclose the definitions.
```suggestion
unless SSL_IVNAMES.include?(:@extra_chain_cert)
  class Net::HTTP
    SSL_IVNAMES << :@extra_chain_cert 
    SSL_ATTRIBUTES << :extra_chain_cert
    attr_accessor :extra_chain_cert
  end
end
```